lesson19/src/js/slider.js

```diff
@@ -0,0 +1,210 @@
+  const slider = createAttributedElements("ul", {class:"slider"});
+  const fragment = document.createDocumentFragment();
+  item.forEach((image, i) => {
+    const sliderItem = createAttributedElements("li", {class:"slider__item"});
+    sliderItem.dataset.slideIndex = i + 1;
+
+    const sliderImage = createAttributedElements("img", {class:"slider__image"});
+    sliderImage.src = image.image;
+    sliderImage.width = image.width;
+    sliderImage.height = image.height;
+
+    i === 0 && sliderItem.classList.add("is-active");
+
+    fragment.appendChild(sliderItem).appendChild(sliderImage);
+  });
+
+  slideImageArea.appendChild(slider).appendChild(fragment);
+}
+
+function renderSlideButton() {
+  const fragment = document.createDocumentFragment();
+  const buttonArea = createAttributedElements("div", {class:"slider-button"});
+
+  const direction = ["previous", "next"];
+  direction.forEach((element) => {
+    const button = document.createElement("button");
+    button.classList.add("arrow", `${element}`);
+    button.id = `js-button_${element}`;
+    button.textContent = `${element}`;
+    button.setAttribute("aria-label", `${element}`);
+    element === 'previous' && (button.disabled = 'true');
+    fragment.appendChild(button);
+  });
+
+  const slideArea = document.getElementById('js-mainvisual-images');
+  slideArea.appendChild(buttonArea).appendChild(fragment);
+}
+
+function renderPagination(item) {
+  const pagination = createAttributedElements("div", {class:"slider-pagination"});
+  const fragment = document.createDocumentFragment();
+  const current = document.createElement("span");
+  current.id = "js-current";
+  current.textContent = 1;
+  fragment.appendChild(current);
+
+  const separation = createAttributedElements("span", {class:"separation"});
+  separation.textContent = "/";
+  fragment.appendChild(separation);
+
+  const total = document.createElement("span");
+  total.id = "js-total";
+  total.textContent = item;
+  fragment.appendChild(total);
+
+  const textArea = document.getElementById("js-mainvisual-textarea");
+  textArea.appendChild(pagination).appendChild(fragment);
+}
+
+function renderPaginationBullet(item){
+  const paginationArea = createAttributedElements("div",{class:"mainvisual-paginationbullet"});
+  paginationArea.id = 'js-pagination';
+  const fragment = document.createDocumentFragment();
+  for(let i = 0; i < item; i++){
+    const paginationBullet = createAttributedElements("span", {class:"paginationbullet__bar"});
+    paginationBullet.style.width = "calc(width / item)";
+    paginationBullet.setAttribute ("role", "button");
+    paginationBullet.dataset.paginationIndex = i + 1;
+    fragment.appendChild(paginationBullet);
+
+    i === 0 && paginationBullet.setAttribute("aria-current","true");
+
+  }
+  const slideImageArea = document.getElementById("js-mainvisual-images-inner");
+  slideImageArea.appendChild(paginationArea).appendChild(fragment);
+}
+
+function addNoimage(){
```

Comment:
Q
```js
addNoimage()
```
Is there any reason why the I is not capitalized?
```js
addNoImage()
```

Reply:
I made a mistake in the function name. This has been fixed!

[f9334f0](https://github.com/nyattsu72/hanson-frontend-cource/pull/27/commits/f9334f09182730dd2e7fad406dd5989ad86005a8)